salt/modules/influx.py

```diff
@@ -262,7 +263,17 @@ def user_exists(
     users = user_list(database, user, password, host, port)
     if not isinstance(users, list):
         return False
-    return name in [u['name'] for u in users]
+
+    for user in users:
+        # the dict key could be different depending on influxdb version
+        username = user.get('user', user.get('name'))
+        if username:
+            if username == name:
+                return True
+        else:
+            log.warning('Could not find username in user: %s', user)
```

Comment:
I think the log.warning should really never happen, which is why I didn't put much thought into whether to put it in the loop or not. In InfluxDB 0.8, `user.get('name')` will return the username, and in 0.9+, `user.get('user')` will return the username. I don't know how 0.7 would behave.

If I put the log outside of the loop, I wouldn't be able to log the user dict that doesn't contain the expected keys, and I'd have to add an additional variable - `found_missing_user_keys` or something.